lib/graphql/tracing/platform_tracing.rb

```diff
@@ -104,19 +104,27 @@ def self.use(schema_defn, options = {})
 
       private
 
-      # Get the transaction name based on the operation type and name
+      # Get the transaction name based on the operation type and name if possible, or fall back to a user provided
+      # one. Useful for anonymous queries.
       def transaction_name(query)
         selected_op = query.selected_operation
         if selected_op
           op_type = selected_op.operation_type
           op_name = selected_op.name || "anonymous"
+        elsif fallback_name = fallback_transaction_name(query.context)
+          return "GraphQL/#{fallback_name}"
         else
           op_type = "query"
           op_name = "anonymous"
         end
         "GraphQL/#{op_type}.#{op_name}"
```

Comment:
sorry, personally, I have a hard time with multi-return methods, because I end up forgetting the different "exits" that the method has. if you don't mind: 

```suggestion
        txn_name = if selected_op
          op_type = selected_op.operation_type
          op_name = selected_op.name || "anonymous"
          "#{op_type}.#{op_name}"
        elsif (fallback_name = fallback_transaction_name(query.context))
          fallback_name
        else
          "query.anonymous"
        end
        "GraphQL/#{txn_name}"
```